CS02-React&Redux.md

```diff
@@ -0,0 +1,127 @@
+
+#### Redux is a predictable state container for JavaScript apps.
+
+- It helps you write applications that behave consistently, run in different environments (client, server, and native), and are easy to test.
+
+### 2. Describe the history of the project -- when / how did it start? Has it grown?
+
+- React was created by Jordan Walke, a software engineer at Facebook. He was influenced by XHP, an HTML component framework for PHP. It was first deployed on Facebook's newsfeed in 2011 and later on Instagram.com in 2012. It was open-sourced at JSConf US in May 2013.'s React.js Conf in February 2015 and open-sourced in March 2015.
+
+- Redux was created by Dan Abramov and Andrew Clark in 2015. Abramov began writing the first Redux implementation while preparing for a conference talk at React Europe on hot reloading. Abramov remarks, "I was trying to make a proof of concept of Flux where I could change the logic. And it would let me time travel. And it would let me reapply the future actions on the code change."
+
+### 3. Who are the contributors to the project?
+
+- Facebook
+- Community
+    - React: 1240 Contributors
+    - ![](https://i.imgur.com/IpgTIFL.png)
+    - Redux: 613 Contributors
+    - ![](https://i.imgur.com/Ew7q7ed.png)
+
+
+### 4. Describe the project's funding. Is it funded? What kinds of support does it get?
+
+[Facebook Open Source](https://opensource.fb.com/)
+
+- Fundings
+- Labours
+- Tech Supports
+- Beer 🍺 & Coffee ☕️
+
+### 5. Describe the project's community. Are there separate developer / user communities? 
+
+- **React**
+    - [React Official Discuss](https://discuss.reactjs.org/) 	18.7k Posts
+    - [Stackoverflow](https://stackoverflow.com/questions/tagged/reactjs) 66.2k Watchers 100.9k Questions
+    - [Reddit](https://www.reddit.com/r/reactjs/) 69.4k Subscribers
+    - Others: 
+    [DEV’s React community](https://dev.to/t/react) | [Hashnode’s React community](https://hashnode.com/n/reactjs) | [Spectrum’s React community](https://spectrum.chat/react)
+
+- **Redux**
+    - [Stackoverflow](https://stackoverflow.com/questions/tagged/redux) 9.6k Watchers 15.4k Questions
+
+- **Reactiflux**
+    - [Homepage](https://www.reactiflux.com/) on Github
+    - [Discord](https://discord.gg/0ZcbPKXt5bZ6au5t) 37.7k Subscribers
+
+### 6. Where are the primary places they interact?
+
+- React:
+    - Facebook -> Github
+- Redux:
+    - Github
+
+### 7. Does the project have a Code of Conduct? What does it consist of? If it does not, why?
+
+- Yes
+- [Facebook Opensource CoD](https://code.fb.com/codeofconduct/)
+- Contents:
+    - Be friendly and patient.
+    - Be welcoming.
+    - Be considerat.
+    - Be respectful.
+    - Be careful in the words that you choose.
+    - We encourage everyone to participate and are committed to building a community for all. (Diversity Statement)
+    - Reporting Issues
+
+### 8. Does the project have a license? Why was the license chosen?
```

Comment:
Could you include some information about the licensing controversy here?

https://en.wikipedia.org/wiki/React_(JavaScript_library)#Licensing_controversy

Reply:
I think it's also worth noting that contributors have to sign a "CLA":

https://code.facebook.com/cla

Reply:
Oh yes, I should mention these on the document!